lib/evaluate/intrinsics.cc

```diff
@@ -976,14 +976,16 @@ std::optional<SpecificCall> IntrinsicInterface::Match(
        dummy[dummyArgPatterns].keyword != nullptr;
        ++dummyArgPatterns) {
   }
-  std::vector<ActualArgument *> actualForDummy(dummyArgPatterns, nullptr);
   // MAX and MIN (and others that map to them) allow their last argument to
   // be repeated indefinitely.  The actualForDummy vector is sized
   // and null-initialized to the non-repeated dummy argument count,
   // but additional actual argument pointers can be pushed on it
   // when this flag is set.
   bool repeatLastDummy{dummyArgPatterns > 0 &&
       dummy[dummyArgPatterns - 1].optionality == Optionality::repeats};
+  std::size_t nonRepeatedDummys{
+      repeatLastDummy ? dummyArgPatterns - 1 : dummyArgPatterns};
+  std::vector<ActualArgument *> actualForDummy(nonRepeatedDummys, nullptr);
```

Comment:
> // The actualForDummy vector is sized
> // and null-initialized to the non-repeated dummy argument count,
> // but additional actual argument pointers can be pushed on it
> // when this flag is set.

I think the comment about `actualForDummy` needs to be modified. Since additional arguments can't be pushed, and the vector is always constructed to the non-repeated dummy argument count.

Reply:
Also, the commit message could mention this change as well.

Reply:
Actually, I think the comment is now enforced (previously, it was not completely true that it was sized to the non-repeated dummy argument count, the repeated dummy argument was included).
Also I am not sure I follow your point that additional arguments can't be pushed.
I think they can be pushed and are. See line: https://github.com/flang-compiler/f18/blob/a4a91f2f9b396020208cf6e536d1d3687e1e7fd6/lib/evaluate/intrinsics.cc#L1030
The commit message mentioned something was fixed in `Match`, but was not explicit about what was the fix. I will name the array explicitly in the commit message (it is in the new commit and will get squashed before I merged).

Reply:
I think I misread the code, what you said makes sense. Thanks.